modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java

```diff
@@ -2718,6 +2736,25 @@ private RestoreLogicalState applyLogicalUpdates(
         for (DatabaseLifecycleListener lsnr : getDatabaseListeners(cctx.kernalContext()))
             lsnr.afterLogicalUpdatesApplied(this, restoreLogicalState);
 
+        List<IgniteInternalFuture<?>> clearFuts = new ArrayList<>();
+
+        partitionRecoveryClearing.forEach((key, value) -> {
+                    CacheGroupContext grp = ctx.cache().cacheGroup(key.getGroupId());
```

Comment:
I think it would be good to check that group context is not `null` before proceeding further.